### ISSUE TITLE
custom script - parallelrunstep not working

### ISSUE THREAD

**Issue by michalmar** [microsoft/solution-accelerator-many-models#118]
when I try run the example `Custom_Script/02_CustomScript_Training_Pipeline.ipynb` I cannot create `ParallelRunConfig` 
```
parallel_run_config = ParallelRunConfig(
    source_directory='./scripts',
    entry_script='train.py',
    mini_batch_size="1",
    run_invocation_timeout=timeout,
    error_threshold=10,
    output_action="append_row",
    environment=train_env,
    process_count_per_node=processes_per_node,
    compute_target=compute,
    node_count=node_count)
```
it gives error:
```
in /anaconda/envs/azureml_py36/lib/python3.6/site-packages/azureml/pipeline/steps/parallel_run_config.py
...
TypeError: __init__() got an unexpected keyword argument 'allowed_failed_count'
```

I have updated to latest SDK (pipeline):
```
zureml-pipeline-core==1.22.0
azureml-pipeline-steps==1.22.0
```

when I downgrade to `1.20.0` it works:
```
zureml-pipeline-core==1.20.0
azureml-pipeline-steps==1.20.0
```

so fix is:
```
!pip install update azureml-pipeline-steps==1.20.0
```

**Comment by dkmiller** [comment]
The official docs for ParallelRunConfig still show that keyword argument: [`azureml.pipeline.steps.ParallelRunConfig`](https://docs.microsoft.com/en-us/python/api/azureml-pipeline-steps/azureml.pipeline.steps.parallelrunconfig?view=azure-ml-py).

I wonder if you're using the stale one &mdash; [`azureml.contrib.pipeline.steps.parallel_run_config.ParallelRunConfig`](https://docs.microsoft.com/en-us/python/api/azureml-contrib-pipeline-steps/azureml.contrib.pipeline.steps.parallel_run_config.parallelrunconfig?view=azure-ml-py)?

**Comment by michalmar** [comment]
@dkmiller how can I check the stale one?

**Comment by dkmiller** [comment]
Look in your script to see from where you are importing the `ParallelRunConfig`.

Also, suggest you make sure to pull the latest version of this repo.

**Comment by michalmar** [comment]
I am using official not staled repo:
`from azureml.pipeline.steps import ParallelRunConfig`

repo cloned couple days ago - so not sure where the problem comes from..

**Comment by dkmiller** [comment]
Thanks for confirming &mdash; I could also repro this in a "clean" Dockerfile:

```dockerfile
FROM python:3.8

RUN pip install azureml-pipeline-steps==1.22.0

RUN python -c "from azureml.pipeline.steps import ParallelRunConfig; cfg = ParallelRunConfig(allow_failed_count=10)"
```

I've raised this internally as an issue with Azure ML's Python SDK team.

**Comment by michalmar** [comment]
@dkmiller I am running on AML CI - should I create new conda env?

**Comment by dkmiller** [comment]
Yes, I'd suggest creating a new Conda environment from scratch. Follow this article to expose that Conda environment as a Jupyter kernel: https://medium.com/@nrk25693/how-to-add-your-conda-environment-to-your-jupyter-notebook-in-just-4-steps-abeab8b8d084 .
